Remove non-essential containerization material

ariostas opened the pull request:
I removed most mentions of containerization throughout the material. When introducing compilation and ROOT scripts, I showed how to get ROOT with Conda. However, for the rest of the material, it makes more sense to run on the ROOT Docker images, so I added a brief paragraph mentioning that it's not necessary to know anything about containerization, and that these images can be thought of as a more convenient set of pre-installed software.

I did remove every mention of `act` since it requires Docker, and it doesn't add much value to the material.

Lastly, I fixed some small issues that were present in the yaml files like outdated action versions, typos, and missing escape sequences.

ariostas committed:
Removed non-essential containerization material

goi42 commented on _episodes/17-github-packages.md:
Should this be updated to reference Podman instead of Docker (following [this issue](https://github.com/hsf-training/hsf-training-docker/issues/81))? Maybe that's not doable in this context--I'm not super familiar with this.

ariostas replied:
Okay, I'll adjust the wording a bit.

goi42 commented on _episodes/08-adding-ci-to-your-code.md:
Going back and forth between Conda and containers seems unnecessary and potentially confusing. I suggest instead just using containers, then adding a brief explanation below:

> ...
> ## No root-config?
> Ok, so maybe we were a little naive here. GitHub runners come pre-installed with a wide variety of software that is commonly needed in CI workflows (e.g. for Ubuntu 22.04 runners the list can be found [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md)). ROOT is not pre-installed, so we will have to add a step to install it ourselves. After reading the [ROOT documentation](https://root.cern/install/#run-in-a-docker-container), we find that a convenient way to run it on Linux is using something called a Docker container.
> 
> For this tutorial you don’t need to know anything about containers. You can just think of them as a base software set that comes pre-installed on the system that runs your code.
> 
> We will be using the Docker images hosted at the [rootproject/root Docker Hub](https://hub.docker.com/r/rootproject/root). Here’s how we can do it, using the image with tag `6.26.10-conda`:
> ```yaml
> build_skim:
>   runs-on: ubuntu-latest
>   container: rootproject/root:6.26.10-conda
>   defaults:
>     run:
>       shell: bash -el {0}
>   steps:
>     - name: build
>       run: |
>         COMPILER=$(root-config --cxx)
>         $COMPILER -g -O3 -Wall -Wextra -Wpedantic -o skim skim.cxx
> ```
> 
> > ## Failed again???
> > ...
> 
> ...
> ## Ways to get software
> As we saw before, GitHub pre-installs many common software packages and libraries that people might need, but often we need to install additional software. There are often actions we can use for this, like `actions/setup-python` to install python or `mamba-org/setup-micromamba` to install [Mamba](https://mamba.readthedocs.io/en/latest/) (an alternative to [Conda](https://docs.conda.io/), an environment manager). These actions are simply repositories that contain scripts to install or perform certain actions. You can find more information about these actions by going to github.com/<name-of-action>. For example, for `mamba-org/setup-micromamba` you can find more information at https://github.com/mamba-org/setup-micromamba.
> 
> If we wanted to use Conda instead of Docker, our `build_skim` job would look like this:
> ```yaml
> build_skim:
>  runs-on: ubuntu-latest
>  defaults:
>    run:
>      shell: bash -el {0}
>  steps:
>    - name: Install ROOT
>      uses: mamba-org/setup-micromamba@v1
>      with:
>        environment-name: env
>        create-args: root
>    - name: checkout repository
>      uses: actions/checkout@v4
>    - name: build
>      run: |
>        COMPILER=$(root-config --cxx)
>        FLAGS=$(root-config --cflags --libs)
>        $COMPILER -g -O3 -Wall -Wextra -Wpedantic -o skim skim.cxx $FLAGS
> ```
> 
> Above, we chose to use a Docker container that came pre-installed with the software we needed. There are several tools that are used for containerization, like Docker, Podman, and Apptainer (formerly Singularity). Again, for this tutorial you don’t need to know anything about containerization, but you can learn more about it in the [dedicated tutorial](https://hsf-training.github.io/hsf-training-docker/).
> 
> ## Building multiple versions
> ...

On a related note, in `05-understanding-yaml-and-ci`, the idea of specifying `runs-on` is briefly introduced as a "virtual environment" without explanation. This concept should be fleshed out a little there (with links to more information).

ariostas replied:
Yeah, that's a good idea. I'll make that change. Thanks!

For the `runs-on` part, I agree that it should be improved. I'll do that too.

ariostas committed:
Small fixes and removed old videos

ariostas commented:
I addressed the comments by @goi42, fixed a few small issues that I found while recording the lessons, and removed the old videos. I'll wait until tomorrow to see if there are any more comments and I'll merge the PR.

goi42 commented on _episodes/05-understanding-yaml-and-ci.md:
```diff
@@ -44,7 +47,7 @@ For more details, check [this link](https://docs.github.com/en/free-pro-team@lat
 > `<job_id>`: Each job must have an id to associate with the job, job_1 in the above example. The key job_id is a string that is unique to the jobs object. It must start with a letter or _ and contain only alphanumeric characters, -, or _. Its value is a map of the job's configuration data.
 >
-> - `runs-on`: **Required**. Each job runs in an virtual environment (runner) specified by the key runs-on. Available environments can be found [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on).
+> - `runs-on`: **Required**. Each job runs in a particular type of machine that we choose with this key. There are options for the major operating systems (Linux, Windows, and macOS) and different versions of them. The available options can be found [here](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on).
```
These are still called "runners" in lesson 8, so it might be useful to retain this bit of jargon here.

ariostas replied:
Okay, sounds good.

goi42 commented on _episodes/08-adding-ci-to-your-code.md:
````diff
@@ -83,32 +87,32 @@ git push -u origin feature/add-actions
         $COMPILER -g -O3 -Wall -Wextra -Wpedantic -o skim skim.cxx
 ```
 
+Note the extra line `container: rootproject/root:6.26.10-conda` that specifies the container image that we want to use. Since it comes pre-packages with ROOT we no longer need to have a step to install it. This image also contains other tools that we will need for the rest of the tutorial.
````
```suggestion
Note the extra line `container: rootproject/root:6.26.10-conda` that specifies the container image that we want to use. Since it comes pre-packaged with ROOT, we do not need to have a step to install it. This image also contains other tools that we will need for the rest of the tutorial.
```

goi42 commented on _episodes/08-adding-ci-to-your-code.md:
```diff
@@ -145,30 +149,34 @@ build_skim:
-## Simplyfing our CI with container images
+## Ways to get software
 
 As we saw before, GitHub pre-installs many common software packages and libraries that people might need, but often we need to install additional software. There are often actions we can use for this, like `actions/setup-python` to install python or `mamba-org/setup-micromamba` to install Mamba (an alternative to Conda). These actions are simply repositories that contain scripts to install or per ...
```
```suggestion
As we saw before, GitHub pre-installs many common software packages and libraries that people might need, but often we need to install additional software. There are often actions we can use for this, like `actions/setup-python` to install python or `mamba-org/setup-micromamba` to install [Mamba](https://mamba.readthedocs.io) (an alternative to [Conda](https://docs.conda.io), an environment manager). These actions are simply repositories that contain scripts to install or perform certain actions. You can find more information about these actions by going to github.com/\<name-of-action\>. For example, for `mamba-org/setup-micromamba` you can find more information at [https://github.com/mamba-org/setup-micromamba](https://github.com/mamba-org/setup-micromamba).
```

ariostas committed:
Fixed typos

Co-authored-by: Michael K. Wilkinson <goi42@users.noreply.github.com>

ariostas committed:
Added links to Mamba and Conda websites

Co-authored-by: Michael K. Wilkinson <goi42@users.noreply.github.com>

ariostas committed:
Introduce runner terminology
